src/fungal_effects.cpp

```diff
@@ -92,200 +92,148 @@ void fungal_effects::create_spores( const tripoint &p, Creature *origin )
     }
 }
 
-bool fungal_effects::marlossify( const tripoint &p )
+void fungal_effects::marlossify( const tripoint &p )
 {
     auto &terrain = m.ter( p ).obj();
     if( one_in( 25 ) && ( terrain.movecost != 0 && !m.has_furn( p ) )
         && !terrain.has_flag( TFLAG_DEEP_WATER ) ) {
         m.ter_set( p, t_marloss );
-        return true;
+        return;
     }
     for( int i = 0; i < 25; i++ ) {
-        if( !spread_fungus( p ) ) {
-            return true;
+        const bool is_fungi = m.has_flag_ter( "FUNGUS", p );
```

Comment:
Do we really need to declare this variable twenty six times?

Reply:
Fixed